### PR TITLE
Alloy Migration

### PR DESCRIPTION
**Pull request by 0xDmtri**
Last bits and pieces were fixed and now we have successfully passed all tests.

### CODE REVIEW

**Comment by 0xDmtri** [comment]
We probably need to bump version too, up to you

**Comment by 0xDmtri** [comment]
Removed `Cargo.lock` file from git as it is a lib (unless you think otherwise lmk)

**Comment by 0xKitsune** [comment]
Sgtm, reviewing this PR today. Thanks again for all the work on this!

File: src/amm/erc_4626/mod.rs [hunk 1]
```diff
@@ -320,16 +320,16 @@ mod tests {
     use super::ERC4626Vault;
 
     #[tokio::test]
-    async fn test_get_vault_data() -> eyre::Result<()> {
-        let rpc_endpoint = std::env::var("ETHEREUM_RPC_ENDPOINT")?;
-        let provider = Arc::new(ProviderBuilder::new().on_http(rpc_endpoint.parse()?));
+    async fn test_get_vault_data() {
```
Comment (0xKitsune) [inline src/amm/erc_4626/mod.rs:323]: Why remove `eyre` for error handling here?

**Review by 0xKitsune — APPROVED** [review]

**Pull request merged**